src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -493,56 +530,134 @@ private Float[][] PostProcess(Float[][] y, Float[] sigma, Float[] z, int d, int
             return y;
         }
 
-        private ColumnType[] InitColumnTypes()
+        protected override IRowMapper MakeRowMapper(ISchema schema) => new Mapper(this, Schema.Create(schema));
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
-            Host.Assert(Infos.Length == _transformInfos.Length);
-            var types = new ColumnType[Infos.Length];
-            for (int i = 0; i < _transformInfos.Length; i++)
-                types[i] = new VectorType(NumberType.Float, _transformInfos[i].Rank);
-            Metadata.Seal();
-            return types;
+            ValidatePcaInput(Host, inputSchema.GetColumnName(srcCol), inputSchema.GetColumnType(srcCol));
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        internal static void ValidatePcaInput(IHost host, string name, ColumnType type)
```

Comment:
>IHost [](start = 46, length = 5)

let's use the narrowest possible interface. In this case it should be `IExceptionContext`

Reply:
done

---
In reply to: [227587653](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227587653) [](ancestors = 227587653)